### DISCUSSION TITLE
Add a way to show only used workspaces in the groupsbox widget.

### DISCUSSION THREAD

**Discussion by AvgBlank** [qtile/qtile#4932] [Ideas] [answered]
Similarly to how i3 does it, make the widget only show workspaces that have apps opened on them, or is the current selected workspace.

**Comment by elParaguayo** [comment, marked as answer]
The main `GroupBox` widget takes an optional parameter called `hide_unused`. Setting that to `hide_unused=True` will probably do what you want.

You can also do this with `GroupBox2` in qtile-extras (needs latest git version). The advantage of this is you can have much more control about the appearance of groups. To get a similar effect, you could do:

```python
from qtile_extras import widget
from qtile_extras.widget.groupbox2 import GroupBoxRule


# in your bar
widget.GroupBox2(
    rules=[
        GroupBoxRule(visible=True).when(focused=True),
        GroupBoxRule(visible=False).when(occupied=False)
    ]
),
```

**Comment by AvgBlank** [comment]
Great, the first solution of setting hide_unused=True worked for me.
